### PR TITLE
makes the PDF container focusable to provide accessibility

### PR DESCRIPTION
**Pull request by mollfpr**
### Details

Adding `focusable` to the PDF container so the scroll with an arrow key is go through the container instead of depending on the `body` tag focusable.

### Fixed Issues

$ https://github.com/Expensify/App/issues/10824   
$ https://github.com/Expensify/App/issues/10824#issuecomment-1298176088

### Tests

**Test on Web/Desktop**

1. Sign in to NewDot with any account
2. Select a chat with any user
3. Click the plus button to the left of the message input, and select "Add attachment"
4. Select a PDF
5. Click send
6. After the PDF uploads, click to open it
7. Close it (notice that tabIndex=-1 on the body HTML element)
8. Click the plus button to the left of the message input, and select "Add attachment"
9. Select a PDF
10. Click on it
11. Click the down arrow key to scroll the PDF down, and verify that it scrolls down
12. Click the up arrow key to scroll the PDF up, and verify that it scrolls up

- [x] Verify that no errors appear in the JS console

### QA Steps

**Test on Web/Desktop**

1. Sign in to NewDot with any account
2. Select a chat with any user
3. Click the plus button to the left of the message input, and select "Add attachment"
4. Select a PDF
5. Click send
6. After the PDF uploads, click to open it
7. Close it (notice that tabIndex=-1 on the body HTML element)
8. Click the plus button to the left of the message input, and select "Add attachment"
9. Select a PDF
10. Click on it
11. Click the down arrow key to scroll the PDF down, and verify that it scrolls down
12. Click the up arrow key to scroll the PDF up, and verify that it scrolls up

- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] iOS / native
    - [x] Android / native
    - [x] iOS / Safari
    - [x] Android / Chrome
    - [x] MacOS / Chrome
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product was added in all `src/languages/*` files
    - [x] I verified any copy / text that was added to the app is correct English and approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

<details>
<summary><h4>PR Reviewer Checklist</h4>

The reviewer will copy/paste it into a new comment and complete it after the author checklist is completed
</summary>

- [ ] I have verified the author checklist is complete (all boxes are checked off).
- [ ] I verified the correct issue is linked in the `### Fixed Issues` section above
- [ ] I verified testing steps are clear and they cover the changes made in this PR
    - [ ] I verified the steps for local testing are in the `Tests` section
    - [ ] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [ ] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [ ] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [ ] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I verified tests pass on **all platforms** & I tested again on:
    - [ ] iOS / native
    - [ ] Android / native
    - [ ] iOS / Safari
    - [ ] Android / Chrome
    - [ ] MacOS / Chrome
    - [ ] MacOS / Desktop
- [ ] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [ ] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [ ] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [ ] I verified that comments were added to code that is not self explanatory
    - [ ] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [ ] I verified any copy / text shown in the product was added in all `src/languages/*` files
    - [ ] I verified any copy / text that was added to the app is correct English and approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [ ] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [ ] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [ ] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [ ] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [ ] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [ ] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [ ] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [ ] If a new component is created I verified that:
    - [ ] A similar component doesn't exist in the codebase
    - [ ] All props are defined accurately and each prop has a `/** comment above it */`
    - [ ] The file is named correctly
    - [ ] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [ ] The only data being stored in the state is data necessary for rendering and nothing else
    - [ ] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [ ] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [ ] All JSX used for rendering exists in the render method
    - [ ] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [ ] If a new CSS style is added I verified that:
    - [ ] A similar style doesn't already exist
    - [ ] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [ ] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [ ] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [ ] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

</details>

#### Web

https://user-images.githubusercontent.com/25520267/199569331-579797c8-efc9-4742-8d9b-f0bd9ef00e52.mov

#### Mobile Web - Chrome

https://user-images.githubusercontent.com/25520267/199569344-b2216887-2369-459c-bbb4-28262b71a3f0.mov

#### Mobile Web - Safari

https://user-images.githubusercontent.com/25520267/199569357-6d0e6a58-1d3b-4d13-946b-a4ce661d8dd9.mov

#### Desktop

https://user-images.githubusercontent.com/25520267/199569377-d5bf90b1-e154-4fdf-88eb-258430b06bcc.mp4

#### iOS

https://user-images.githubusercontent.com/25520267/199569390-3ca7b70e-f205-42f1-9901-39c5db680564.mov

#### Android

https://user-images.githubusercontent.com/25520267/199569416-423be05d-2a81-4714-8185-e062fc9cc04d.mov

### CODE REVIEW

**Comment by Santhosh-Sellavel** [comment]
@mollfpr 
Can we update the testing steps, please remove the first two steps with the following.
1. Open New Expensify App & login

**Comment by Santhosh-Sellavel** [comment]
<details>
<summary><h4>Screen Recordings</h4></summary>

#### Desktop & Web
https://user-images.githubusercontent.com/85645967/200201237-b1ef569b-257e-49cc-a432-19de8ce2e2e5.mov

#### Android - iOS 
https://user-images.githubusercontent.com/85645967/200201295-a8598fdd-6785-4fc4-89c3-39f9c97e0fee.mov

#### Android - iOS mWeb
https://user-images.githubusercontent.com/85645967/200201248-3fdb7e27-4911-4c35-a9b2-9de038d52fff.mov

</details>

**Comment by Santhosh-Sellavel** [comment]
<details>
<summary><h4>PR Reviewer Checklist</h4>

The reviewer will copy/paste it into a new comment and complete it after the author checklist is completed
</summary>

- [ ] I have verified the author checklist is complete (all boxes are checked off).
- [ ] I verified the correct issue is linked in the `### Fixed Issues` section above
- [ ] I verified testing steps are clear and they cover the changes made in this PR
    - [ ] I verified the steps for local testing are in the `Tests` section
    - [ ] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [ ] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [ ] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [ ] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I verified tests pass on **all platforms** & I tested again on:
    - [ ] iOS / native
    - [ ] Android / native
    - [ ] iOS / Safari
    - [ ] Android / Chrome
    - [ ] MacOS / Chrome
    - [ ] MacOS / Desktop
- [ ] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [ ] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [ ] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [ ] I verified that comments were added to code that is not self explanatory
    - [ ] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [ ] I verified any copy / text shown in the product was added in all `src/languages/*` files
    - [ ] I verified any copy / text that was added to the app is correct English and approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [ ] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [ ] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [ ] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [ ] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [ ] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [ ] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [ ] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [ ] If a new component is created I verified that:
    - [ ] A similar component doesn't exist in the codebase
    - [ ] All props are defined accurately and each prop has a `/** comment above it */`
    - [ ] The file is named correctly
    - [ ] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [ ] The only data being stored in the state is data necessary for rendering and nothing else
    - [ ] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [ ] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [ ] All JSX used for rendering exists in the render method
    - [ ] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [ ] If a new CSS style is added I verified that:
    - [ ] A similar style doesn't already exist
    - [ ] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [ ] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [ ] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [ ] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

</details>

**Comment by Santhosh-Sellavel** [comment]
@neil-marcellini 
While Testing offline, I found this error. Should I log a bug for this? As this is send attachment flow and I expect a preview here
Offline-Issue

**Review by Santhosh-Sellavel — COMMENTED** [review]
@neil-marcellini 
Just waiting for @mollfpr to update the test steps & clarification on the above [bug](https://github.com/Expensify/App/pull/12391#issuecomment-1304926841)!

**Comment by mollfpr** [comment]
@Santhosh-Sellavel updated testing step!

How do you repro the bug? it’s not happen to me

**Comment by Santhosh-Sellavel** [comment]
https://user-images.githubusercontent.com/85645967/200205740-d1d90f4b-358b-46b2-87d0-5ecdc036711e.mov

**Comment by Santhosh-Sellavel** [comment]
> @neil-marcellini While Testing offline, I found this error. Should I log a bug for this? As this is send attachment flow and I expect a preview here Offline-Issue

As this is not introduced here, I reported it in slack [here](https://expensify.slack.com/archives/C01GTK53T8Q/p1667782878782179)

**Review by Santhosh-Sellavel — APPROVED** [review]
Recordings: https://github.com/Expensify/App/pull/12391#issuecomment-1304925780
Checklist: https://github.com/Expensify/App/pull/12391#issuecomment-1304925885
C+ Reviewed
🎀 👀 🎀 

Looks good, tests well!

All you @neil-marcellini

**Review by neil-marcellini — CHANGES_REQUESTED** [review]
>**Test on Web/Desktop**
> 
> 1. Open New Expensify App & login(if not already)
> 2. Login with an Expensify account
> 3. Select any user
> 4. Click the plus button to the left of the message input, and select "Add attachment"
> 5. Select a password-protected PDF
> 6. Verify that the password form appears
> 7. Enter the password and Click Confirm
> 8. Click on the PDF document and use the up and down keys for scrolling
> 9. Verify the scroll is working with the arrow key

I will approve this PR once we update these test steps. The problem is that the tests pass on staging already. See the video below.

https://user-images.githubusercontent.com/26260477/200434579-0f9761ac-4c30-4146-9b6a-dd2cf46b231d.mov

In the video posted in the issue description we can see that at step 7 the user types the password and then hits the Enter key. In the next step they expect the up and down arrow keys to scroll the document. I don't think that's actually a problem, but it's also not what you described with these test steps.

Sorry for the confusion. I think these are the test / QA steps that should actually be used, since we agreed that this is the real issue [here](https://github.com/Expensify/App/issues/10824#issuecomment-1292431822). @Santhosh-Sellavel unfortunately I think you will also have to test on all platforms again.

1. Sign into NewDot with any account
2. Select a chat with any user
3. Click the plus button to the left of the message input, and select "Add attachment"
4. Select a PDF
5. Click send
6. After the PDF uploads, click to open it
7. Close it (notice that `tabIndex=-1` on the body html element)
8. Click the plus button to the left of the message input, and select "Add attachment"
9. Select a PDF
10. Click on it
11. Click the down arrow key to scroll the PDF down, verify that it scrolls down
12. Click the up arrow key to scroll the PDF up, verify that it scrolls up

**Comment by Santhosh-Sellavel** [comment]
@neil-marcellini Web/Desktop alone will have to be enough!

**Comment by mollfpr** [comment]
Thanks @neil-marcellini for the correction, now the test steps is updated.

**Comment by neil-marcellini** [comment]
@Santhosh-Sellavel can you test today please?

**Comment by Santhosh-Sellavel** [comment]
Yeah, will be testing it shortly!

File: src/components/PDFView/index.js [hunk 1]
```diff
@@ -114,12 +114,13 @@ class PDFView extends Component {
         // If we're requesting a password then we need to hide - but still render -
         // the PDF component.
         const pdfContainerStyle = this.state.shouldRequestPassword
-            ? [styles.PDFView, this.props.style, styles.invisible]
-            : [styles.PDFView, this.props.style];
+            ? [styles.PDFView, styles.noSelect, this.props.style, styles.invisible]
+            : [styles.PDFView, styles.noSelect, this.props.style];
```
Comment (mollfpr) [inline src/components/PDFView/index.js:118]: @Santhosh-Sellavel There should be no outline showing when the div is focused. I can't reproduce that outline and neither your first video test.

**Comment by neil-marcellini** [comment]
I'll test now myself.

**Comment by neil-marcellini** [comment]
@Santhosh-Sellavel you must have had some kind of glitch when testing this branch. It works fine for me!

Web

https://user-images.githubusercontent.com/26260477/201196944-7529dcfd-bfe0-4e15-b7f1-7362e76d6074.mov

Desktop

https://user-images.githubusercontent.com/26260477/201198589-6fdbcadf-c491-4b89-adbb-174938c40423.mov

**Review by neil-marcellini — APPROVED** [review]

**Comment by neil-marcellini** [comment]
All checks passed. This is a bug that will be fixed soon.

**Comment by Santhosh-Sellavel** [comment]
Seems something is fishy It's not working for me properly sometimes!

**Comment by JmillsExpensify** [comment]
Echoing @sobitneupane in the [main issue](https://github.com/Expensify/App/issues/10824#issuecomment-1313751312), this PR has caused a regression. Per the BZ checklist, is there anything we can add to TestRail or the PR reviewer checklist to prevent such a regression for happening again?

**Pull request merged**